src/CrudPanel.php

```diff
@@ -105,14 +108,16 @@ public function getModel()
         return $this->model;
     }
 
-    /**
-     * Get the database connection, as specified in the .env file or overwritten by the property on the model.
-     */
-    private function getSchema()
+    protected function getDbColumns()
     {
-        return \Schema::setConnection($this->getModel()->getConnection());
-    }
+        if (empty($this->db_columns)) {
+            $table         = $this->model->getTable();
+            $conn          = $this->model->getConnection();
+            $this->db_columns = $conn->getDoctrineSchemaManager()->listTableColumns($table);
```

Comment:
Alignment is off by one here 😸

Reply:
"There are two hard things in computer science: cache invalidation, naming things, and off-by-one errors."
That's off by two. 😉